be/src/vec/exec/join/vhash_join_node.cpp

```diff
@@ -711,45 +766,62 @@ Status HashJoinNode::get_next(RuntimeState* state, Block* output_block, bool* eo
     Status st;
 
     if (_probe_index < _probe_block.rows()) {
-        MutableBlock mutable_block = (output_block->mem_reuse() && !_have_other_join_conjunct) ?
-                MutableBlock(output_block) :
-                MutableBlock(VectorizedUtils::create_empty_columnswithtypename(
-                        !_have_other_join_conjunct ? row_desc() : _row_desc_for_other_join_conjunt));
-
         std::visit(
-                [&](auto&& arg) {
+            [&](auto&& arg, auto have_other_join_conjunct, 
+            auto is_left_semi_join, auto is_left_anti_join, auto is_inner_join) {
```

Comment:
why keep so many bool, maybe you should use std::variant<std::monostate, Inner Join, Semi Join, Anti Join

Reply:
these three parameter is std::integral_constant, not bool. use std::variant, we also need to distinguish it in next execution. and i think std::variant is not efficient.